cpu/stm32/stmclk/stmclk_l4wb.c

```diff
@@ -151,50 +222,86 @@ void stmclk_init_sysclk(void)
-    while (!(RCC->CR & RCC_CR_HSERDY)) {}
-#endif
+    /* Enable the HSE clock only when it's provided by the board and required:
+        - Use HSE as system clock
+        - Use HSE as PLL input clock
+    */
+    if (IS_ACTIVE(CONFIG_BOARD_HAS_HSE) &&
+        (IS_ACTIVE(CONFIG_USE_CLOCK_HSE) || IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_HSE))) {
+        RCC->CR |= (RCC_CR_HSEON);
+        while (!(RCC->CR & RCC_CR_HSERDY)) {}
+    }
 
-#if ((CLOCK_HSE == 0) || CLOCK_MSI_ENABLE)
-    /* reset clock to MSI with 48MHz, disables all other clocks */
+    if (IS_ACTIVE(CONFIG_USE_CLOCK_HSE)) {
+        /* Select HSE as system clock and configure the different prescalers */
+        RCC->CFGR &= ~RCC_CFGR_SW;
+        RCC->CFGR |= RCC_CFGR_SW_HSE;
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_MSI)) {
 #if defined(CPU_FAM_STM32WB)
-    RCC->CR |= (RCC_CR_MSIRANGE_11 | RCC_CR_MSION);
+        RCC->CR |= (CLOCK_MSIRANGE | RCC_CR_MSION);
 #else
-    RCC->CR |= (RCC_CR_MSIRANGE_11 | RCC_CR_MSION | RCC_CR_MSIRGSEL);
-#endif
-    while (!(RCC->CR & RCC_CR_MSIRDY)) {}
-    /* select the MSI clock for the 48MHz clock tree (USB, RNG) */
-    RCC->CCIPR = (RCC_CCIPR_CLK48SEL_0 | RCC_CCIPR_CLK48SEL_1);
-#if (CLOCK_MSI_LSE_PLL && CLOCK_LSE)
-    /* configure the low speed clock domain */
-    stmclk_enable_lfclk();
-    /* now we can enable the MSI PLL mode to enhance accuracy of the MSI*/
-    RCC->CR |= RCC_CR_MSIPLLEN;
-    while (!(RCC->CR & RCC_CR_MSIRDY)) {}
-#endif /* (CLOCK_MSI_LSE_PLL && CLOCK_LSE) */
-#endif /* ((CLOCK_HSE == 0) || CLOCK_MSI_ENABLE) */
-
-    /* now we can safely configure and start the PLL */
-    RCC->PLLCFGR = (PLL_SRC | PLL_M | PLL_N | PLL_R | RCC_PLLCFGR_PLLREN);
-    RCC->CR |= (RCC_CR_PLLON);
-    while (!(RCC->CR & RCC_CR_PLLRDY)) {}
-
-    /* now that the PLL is running, we use it as system clock */
-    RCC->CFGR |= RCC_CFGR_SW_PLL;
-    while ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_PLL) {}
-
-    stmclk_disable_hsi();
-    irq_restore(is);
+        RCC->CR |= (CLOCK_MSIRANGE | RCC_CR_MSION | RCC_CR_MSIRGSEL);
+#endif
+        while (!(RCC->CR & RCC_CR_MSIRDY)) {}
 
-#ifdef MODULE_PERIPH_RTT
-    /* Ensure LPTIM1 clock source (LSI or LSE) is correctly reset when initializing
-       the clock, this is particularly useful after waking up from deep sleep */
-#if CLOCK_LSE
-    RCC->CCIPR |= RCC_CCIPR_LPTIM1SEL_0 | RCC_CCIPR_LPTIM1SEL_1;
+        if (CONFIG_CLOCK_MSI == MHZ(48)) {
+            /* select the MSI clock for the 48MHz clock tree (USB, RNG) */
+            RCC->CCIPR = (RCC_CCIPR_CLK48SEL_0 | RCC_CCIPR_CLK48SEL_1);
+        }
+        /* Select MSI as system clock and configure the different prescalers */
+        RCC->CFGR = (RCC_CFGR_SW_MSI | CLOCK_AHB_DIV | CLOCK_APB1_DIV | CLOCK_APB2_DIV);
+    }
+    else if (IS_ACTIVE(CONFIG_USE_CLOCK_PLL)) {
+        if (IS_ACTIVE(CONFIG_CLOCK_PLL_SRC_MSI)) {
+            /* reset clock to MSI with 48MHz, disables all other clocks */
+#if defined(CPU_FAM_STM32WB)
```

Comment:
Could `IS_ACTIVE` be used here instead?

Reply:
Not by using a regular C `if` because this is related to the underlying hardware and will fail if not build for an stm32wb.
If it's about using it as a pre-processor check, I'm not sure, there's no use like this in the code base.